packages/core/admin/admin/src/pages/AuthPage/index.js

```diff
@@ -238,6 +240,12 @@ const AuthPage = ({ hasAdmin, setHasAdmin }) => {
     }
   };
 
+  const redirectToPreviousLocation = () => {
+    const locationBeforeAuthenticated = decodeURIComponent(query.get('redirectTo'));
+    const redirectUrl = locationBeforeAuthenticated || '/';
+    push(redirectUrl);
+  };
```

Comment:
I think we need to verify here, that the `redirectTo` URL actually belongs to the Strapi instance, to avoid the case where an attacker may send a malformed link like: `strapi.io/auth/login?redirecTo=attackersWebsite.com`

In Wordpress land the function is called `wp_validate_redirect()` https://developer.wordpress.org/reference/functions/wp_validate_redirect/.

What do you think about that? cc @soupette

Reply:
@gu-stav I don't think we need to check if `redirectTo` belongs to Strapi instance. The reason is that we use `history.push`, which uses `window.location.pushState` underlying. And `window.location.pushState` makes sure the URL must be of the same origin as the current URL.
Reference: https://developer.mozilla.org/en-US/docs/Web/API/History/pushState#:~:text=The%20new%20URL%20must%20be%20of%20the%20same%20origin%20as%20the%20current%20URL%3B

Reply:
I attached the gif of that case.
![Mar-14-2022 19-01-29](https://user-images.githubusercontent.com/8603085/158168183-454c6f1b-1be7-436d-b8cf-81b95f8ab30f.gif)

Reply:
Yes we cannot push to another domain.